launch/launch/launch_description.py

```diff
@@ -124,3 +138,8 @@ def add_entity(self, entity: LaunchDescriptionEntity) -> None:
     def add_action(self, action: Action) -> None:
         """Add an action to the LaunchDescription."""
         self.add_entity(action)
+
+    @property
+    def deprecated(self) -> Optional[Text]:
+        """Getter for deprecated."""
```

Comment:
This is a bit odd for me, as I'd expect `ld.deprecated` to return a boolean, and `ld.deprecation_reason` to return `None` if no deprecated or a string if deprecated.

If we leave it as-is, I think you should put into the docstring that `None` means it is not deprecated, and that a string means it is deprecated (and also what the string is, i.e. the description of why it is deprecated).